specs/Extensions.md

```diff
@@ -0,0 +1,591 @@
+
+                    wil::unique_cotaskmem_string id;
+                    wil::unique_cotaskmem_string name;
+                    BOOL enabled = false;
+                    std::wstring message;
+
+                    extension->get_IsEnabled(&enabled);
+                    extension->get_Id(&id);
+                    extension->get_Name(&name);
+                    extensionIdString = id.get();
+
+                    if (extensionIdString.compare(m_extensionId) == 0)
+                    {
+                        extensionInstalled = true;
+                        break;
+                    }
+                }
+
+                if (!extensionInstalled)
+                {
+                    CHECK_FAILURE(profile6->AddBrowserExtension(
+                        extension_path.c_str(),
+                        Callback<ICoreWebView2ProfileAddBrowserExtensionCompletedHandler>(
+                            [](HRESULT error,
+                               ICoreWebView2BrowserExtension* extension) -> HRESULT
+                            {
+                                if (error != S_OK)
+                                {
+                                    ShowFailure(error, L"AddExtension failed");
+                                    return S_OK;
+                                }
+                                return S_OK;
+                            })
+                            .Get()));
+                }
+                return S_OK;
+            })
+            .Get());
+}
+```
+
+### .NET and WinRT
+```c#
+string m_defaultExtensionId = "123";
+string m_defaultExtensionFolderPath = "//Mydrive";
+
+/// Create WebView Environment with option
+void CreateEnvironmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.AreBrowserExtensionsEnabled = true;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(CoreWebView2EnvironmentOptions options: options);
+    webview.EnsureCoreWebView2Async(environment);
+
+    _ = InstallDefaultExtensions();
+}
+
+// Install default extension before first navigation
+private async System.Threading.Tasks.Task InstallDefaultExtensions()
+{
+    List<CoreWebView2BrowserExtension> extensionsList = await m_coreWebView2.Profile.GetBrowserExtensionsAsync();
+    bool found = false;
+    for (int i = 0; i < extensionsList.Count; ++i)
+    {
+        if (extensionsList[i].Id == _defaultExtensionId)
+        {
+            return;
+            // ... Navigate to first page
+        }
+        else
+        {
+            CoreWebView2Extension extension = await m_coreWebView2.Profile.AddBrowserExtensionAsync(m_defaultExtensionFolderPath);
+        }
+    }
+}
+```
+
+## End user manages extensions
+
+Provide end user the ability to add, remove, enable and disable a browser extension from profile.
+
+### Win32 C++
+```cpp
+void ScriptComponent::AddBrowserExtension()
```

Comment:
The sample code between Win32 and .NET/WinRT should match. They should have same method names and code within.

Reply:
I made some change to naming and structure. Let me know if it works